fuel/streams.py

```diff
@@ -135,3 +138,42 @@ def get_epoch_iterator(self, **kwargs):
         else:
             self._fresh_state = False
         return super(DataStream, self).get_epoch_iterator(**kwargs)
+
+
+class ServerDataStream(AbstractDataStream):
```

Comment:
We should file a ticket about querying `num_examples` and `num_batches` from data streams in general, and over the network in particular...